src/avram/callbacks.cr

```diff
@@ -45,20 +48,57 @@ module Avram::Callbacks
   # You can set defaults, validate, or perform any other setup necessary for
   # saving.
   #
+  # Optionally you can specify a symbol to `if` or `unless` which allows you to
+  # run this conditionally. The symbol should reference a method you've defined
+  # that returns a truthy/falsey value
+  #
   # ```
-  # before_save do
+  # before_save(unless: :skip_callback?) do
   #   validate_required name, age
   # end
+  #
+  # private def skip_callback?
+  #   false
+  # end
   # ```
-  macro before_save
+  macro before_save(if _if = nil, unless _unless = nil)
+    {% if _if != nil && _unless != nil %}
+      {% raise "Your before_save callbacks should only specify `if` or `unless`, but not both." %}
+    {% end %}
+    {% unless _if.is_a?(SymbolLiteral) || _if.is_a?(NilLiteral) %}
```

Comment:
Need to add this for unless as well, and in the other macros that are missing it